server/chat.ts

```diff
@@ -108,6 +108,7 @@ import {formatText, linkRegex, stripFormatting} from './chat-formatter';
 
 // @ts-ignore no typedef available
 import ProbeModule = require('probe-image-size');
+import {ExtendedGroup} from './user-groups';
```

Comment:
I would call this `AuthLevel` and also define `room.settings.modchat` and `Config.whatevermodchat` as it, so they don't need to be cast.